stellar_rust_sdk/src/trade_aggregations/trade_aggregations_request.rs

```diff
@@ -0,0 +1,374 @@
+    /// An alphanumeric 12 asset_type type. It holds an Asset struct with asset code and asset issuer.
+    Alphanumeric12(AssetData),
+}
+
+/// Represents the absense of a resolution value.
+#[derive(Default, Clone)]
+pub struct NoResolution;
+
+/// Represents the resolution value. It can contain a [`ResolutionData`] enum type.
+#[derive(PartialEq, Debug, Default, Clone)]
+pub struct Resolution(pub ResolutionData);
+
+/// Represents the supported segment duration times in milliseconds. 
+#[derive(PartialEq, Debug, Default, Clone)]
+pub enum ResolutionData {
+    #[default]
+    Duration60000,
+    Duration300000,
+    Duration900000,
+    Duration3600000,
+    Duration604800000,
+}
+
+impl std::fmt::Display for ResolutionData {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            ResolutionData::Duration60000 => write!(f, "60000"), // 1 minute
+            ResolutionData::Duration300000 => write!(f, "300000"), // 5 minutes
+            ResolutionData::Duration900000 => write!(f, "900000"), // 15 minutes
+            ResolutionData::Duration3600000 => write!(f, "3600000"), // 1 day
+            ResolutionData::Duration604800000 => write!(f, "604800000"), // 1 week
+        }
+    }
+}
+
+/// Represents a request to list trade aggregations from the Stellar Horizon API.
+///
+/// This structure is used to construct a query to retrieve a comprehensive list of trade aggregations, which will be filtered
+/// by the mandatory base asset, counter asset and resolution fields. Additional filters such as start time, end time, limit
+/// and order can be set. It adheres to the structure and parameters required by the Horizon API for retrieving a 
+/// <a href="https://developers.stellar.org/docs/data/horizon/api-reference/list-trade-aggregations">list of trade aggregations</a>.
+///
+/// # Usage
+///
+/// Create an instance of this struct and set the desired query parameters to filter the list of trade aggregations.
+/// Pass this request object to the [`HorizonClient::get_trade_aggregations`](crate::horizon_client::HorizonClient::get_trade_aggregations)
+/// method to fetch the corresponding data from the Horizon API.
+///
+/// # Example
+/// ```
+/// use stellar_rs::{trade_aggregations::prelude::*, models::*, Paginatable};
+///
+/// let request = TradeAggregationsRequest::new()
+///     .set_base_asset(AssetType::Native).unwrap()
+///     .set_counter_asset(AssetType::Alphanumeric4(AssetData{
+///        asset_issuer: "GBZXN7PIRZGNMHGA7MUUUF4GWPY5AYPV6LY4UV2GL6VJGIQRXFDNMADI".to_string(),
+///        asset_code: "XETH".to_string(),
+///     })).unwrap()
+///     .set_resolution(Resolution(ResolutionData::Duration604800000)).unwrap()
+///     .set_limit(100).unwrap() // Optional limit for response records
+///     .set_order(Order::Desc); // Optional order of records
+///
+/// // Use with HorizonClient::get_trade_aggregations
+/// ```
+///
+#[derive(Clone, Debug, PartialEq, Default)]
```

Comment:
Kunnen we hier de Pagination derive gebruiken? Dan hoef je ook niet de set_limit, set_order te implementeren. Dat regelt de derive voor je.

Reply:
Dat heb ik inderdaad overwogen, alleen ontbreekt `cursor` in deze request. Deze zou dan toegevoegd moeten worden, zonder dat deze gebruikt wordt. Is dat wenselijk?

Reply:
Interessant, want hoe indexed deze paginering nu zonder cursor?

Reply:
Dat kan ik nergens terugvinden in de documentatie. Misschien was een limiet en sorteervolgorde voldoende bij dit endpoint.

Reply:
Ik snap m, het gaat met name om start en end tijd, icm met de resolution. 
In dat geval zou ik zeggen, laat m zoals ie nu is. De implementatie werkt anders dan de andere paginatie functies. DRY is een fabel ;-)